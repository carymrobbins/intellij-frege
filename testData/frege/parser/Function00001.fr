module Function00001 where

foo :: Int -> Int -> Int
foo x y = x + y

bar :: Float -> String -> (Float, String)
bar = (,)

baz :: (a -> b) -> (b -> c) -> (a -> c)
baz f g x = f $ g $ x
