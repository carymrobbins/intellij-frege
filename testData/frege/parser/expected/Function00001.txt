Frege file
  MODULE_DECL
    PsiElement(MODULE)('module')
    MODULE_REF
      MODULE_NAME
        PsiElement(IDENT)('Function00001')
    PsiElement(WHERE)('where')
  PsiElement(EOL)('\n')
  PsiElement(IDENT)('foo')
  PsiElement(BAD_CHARACTER)(':')
  PsiElement(BAD_CHARACTER)(':')
  PsiElement(IDENT)('Int')
  PsiElement(OPERATOR)('-')
  PsiElement(OPERATOR)('>')
  PsiElement(IDENT)('Int')
  PsiElement(OPERATOR)('-')
  PsiElement(OPERATOR)('>')
  PsiElement(IDENT)('Int')
  PsiElement(EOL)('\n')
  PsiElement(IDENT)('foo')
  PsiElement(IDENT)('x')
  PsiElement(IDENT)('y')
  PsiElement(=)('=')
  PsiElement(IDENT)('x')
  PsiElement(OPERATOR)('+')
  PsiElement(IDENT)('y')
  PsiElement(EOL)('\n')
  PsiElement(IDENT)('bar')
  PsiElement(BAD_CHARACTER)(':')
  PsiElement(BAD_CHARACTER)(':')
  PsiElement(IDENT)('Float')
  PsiElement(OPERATOR)('-')
  PsiElement(OPERATOR)('>')
  PsiElement(IDENT)('String')
  PsiElement(OPERATOR)('-')
  PsiElement(OPERATOR)('>')
  PsiElement(()('(')
  PsiElement(IDENT)('Float')
  PsiElement(,)(',')
  PsiElement(IDENT)('String')
  PsiElement())(')')
  PsiElement(EOL)('\n')
  PsiElement(IDENT)('bar')
  PsiElement(=)('=')
  PsiElement(()('(')
  PsiElement(,)(',')
  PsiElement())(')')
  PsiElement(EOL)('\n')
  PsiElement(IDENT)('baz')
  PsiElement(BAD_CHARACTER)(':')
  PsiElement(BAD_CHARACTER)(':')
  PsiElement(()('(')
  PsiElement(IDENT)('a')
  PsiElement(OPERATOR)('-')
  PsiElement(OPERATOR)('>')
  PsiElement(IDENT)('b')
  PsiElement())(')')
  PsiElement(OPERATOR)('-')
  PsiElement(OPERATOR)('>')
  PsiElement(()('(')
  PsiElement(IDENT)('b')
  PsiElement(OPERATOR)('-')
  PsiElement(OPERATOR)('>')
  PsiElement(IDENT)('c')
  PsiElement())(')')
  PsiElement(OPERATOR)('-')
  PsiElement(OPERATOR)('>')
  PsiElement(()('(')
  PsiElement(IDENT)('a')
  PsiElement(OPERATOR)('-')
  PsiElement(OPERATOR)('>')
  PsiElement(IDENT)('c')
  PsiElement())(')')
  PsiElement(EOL)('\n')
  PsiElement(IDENT)('baz')
  PsiElement(IDENT)('f')
  PsiElement(IDENT)('g')
  PsiElement(IDENT)('x')
  PsiElement(=)('=')
  PsiElement(IDENT)('f')
  PsiElement(OPERATOR)('$')
  PsiElement(IDENT)('g')
  PsiElement(OPERATOR)('$')
  PsiElement(IDENT)('x')